File: leetcode/1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> answer;
        
        for (int i {0}; i < nums.size(); ++i)
        {
            if (answer.find(target - nums[i]) != answer.end())
            {
                return { answer[target - nums[i]], i };
            }
            
            answer[nums[i]] = i;
        }
        
        return { 0, -1 };
    }
};

File: leetcode/python solutions/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]: # type: ignore
        map = {}

        for i in range(len(nums)):
            if target - nums[i] in map:
                return [map[target - nums[i]], i]
            else:
                map[nums[i]] = i
# @lc code=end

